docs/runtimes/mlrun_code_annotations.ipynb

```diff
@@ -0,0 +1,1253 @@
+       "      <td></td>\n",
+       "      <td></td>\n",
+       "      <td><div class=\"dictlist\">return=hello from part-cell-function</div></td>\n",
+       "      <td></td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div></div>\n",
+       "  <div id=\"result494dc07b-pane\" class=\"right-pane block hidden\">\n",
+       "    <div class=\"pane-header\">\n",
+       "      <span id=\"result494dc07b-title\" class=\"pane-header-title\">Title</span>\n",
+       "      <span onclick=\"closePanel(this)\" paneName=\"result494dc07b\" class=\"close clickable\">&times;</span>\n",
+       "    </div>\n",
+       "    <iframe class=\"fileview\" id=\"result494dc07b-body\"></iframe>\n",
+       "  </div>\n",
+       "</div>\n"
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    },
+    {
+     "data": {
+      "text/html": [
+       "<b> > to track results use the .show() or .logs() methods  or <a href=\"https://dashboard.default-tenant.app.alonm32.iguazio-cd0.com/mlprojects/default/jobs/monitor/5426e665c7bc4ba492e0a704c5555fb6/overview\" target=\"_blank\">click here</a> to open in UI</b>"
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "> 2021-11-01 07:43:14,628 [info] run executed, status=completed\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "<mlrun.model.RunObject at 0x7f3fc9a2bf50>"
+      ]
+     },
+     "execution_count": 14,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "my_multi_section_function = code_to_function('part-cell-function', kind='job')\n",
+    "my_multi_section_function.run(name='part-cell-function', handler='handler', local=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Notes\n",
+    "- Make sure not to have consecutive `# mlrun: start-code` without `# mlrun: end-code` in between them and vice versa.\n",
+    "- Only 1 MLRun function can use nameless annotations per notebook.\n",
+    "- Do not use multiple `# mlrun: start-code` nor multiple `# mlrun: end-code` annotations in a single code cell, only the 1st appearance of each will be used.\n",
+    "- You may use a single annotation:\n",
+    "    1. using `# mlrun: start-code` alone will include code blocks from the annotation to the end of the notebook.\n",
+    "    2. using `# mlrun: end-code` alone will include code blocks from the beginning of the notebook to the annotation."
```

Comment:
This text requires some work, please consult with Jason or Jill how to phrase this. For example, in the first bullet point we should state that nested annotation blocks are not supported.